### PR TITLE
Add privacy manifest.

### PR DESCRIPTION
**Pull request by ykjchen**
https://github.com/google/promises/issues/225

I don't see any documentation to add this to BUILD.bazel, so leaving it out for now.

### CODE REVIEW

**Review by chuckhays — APPROVED** [review]
LGTM

File: Sources/Resources/PrivacyInfo.xcprivacy
Comment (nabs-m) [inline Sources/Resources/PrivacyInfo.xcprivacy:12]: This is an empty privacy manifest, so it's not really declaring anything. You don't need anything other than `NSPrivacyAccessedAPITypes`, so you can remove the others.
Under `NSPrivacyAccessedAPITypes`, you'll need to add the `NSPrivacyAccessedAPIType` and `NSPrivacyAccessedAPITypeReasons` as described [here](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api), based on the usage of each sensitive API.

File: PromisesObjC.podspec
Comment (ncooke3) [inline PromisesObjC.podspec:26]: I _think_ this should instead be packaged via the `resource_bundle` API instead. I'm experimenting to see how Firebase's release tooling handles `resource`.

Comment (ykjchen) [reply PromisesObjC.podspec:26]: Thanks for looking into it. I did see other podspecs where this was included as a `resource`:
https://github.com/treasure-data/td-ios-sdk/blob/master/TreasureData-iOS-SDK.podspec
https://github.com/getsentry/sentry-cocoa/blob/main/Sentry.podspec

but I'll wait for your confirmation.

File: Sources/Resources/PrivacyInfo.xcprivacy
Comment (nabs-m) [reply Sources/Resources/PrivacyInfo.xcprivacy:12]: > Since there is no usage of sensitive APIs should that just be an empty array?

Do you mean that none of the [sensitive APIs](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api) are used at all? Or they are used but fall within the accepted reasons? If it's the former, yes you wouldn't list anything under `NSPrivacyAccessedAPITypes`, but if it's the latter, you'd still want to specify which ones and their reasons.

> I set them all to empty values in order to be explicit, since I saw a comment saying that at least NSPrivacyTracking needed to be included: https://github.com/CocoaPods/CocoaPods/issues/11990#issuecomment-1855075327

I don't believe that's true. Check the note in [this doc](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files):
`You only need to supply NSPrivacyAccessedAPITypes for apps and third-party SDKs on iOS, iPadOS, tvOS, visionOS, and watchOS.`
So there's no need to include it there, I'd keep the `xcprivacy` file focused on the sensitive APIs.

File: PromisesObjC.podspec
Comment (nabs-m) [reply PromisesObjC.podspec:26]: I've seen all 3 combinations work:
* `s.resource`
* `s.resources`
* s.resource_bundle`

Since this is a singular file for now, I'd keep it as `resource` (although `resource_bundle` should work too).

File: Sources/Resources/PrivacyInfo.xcprivacy
Comment (ykjchen) [reply Sources/Resources/PrivacyInfo.xcprivacy:12]: Indeed, no sensitive APIs are used at all.

I feel like that note could be interpreted to mean: "on iOS, etc. you need to supply NSPrivacyAccessedAPITypes; on other OS's you don't need to supply it". Given the ambiguity I'll keep these keys here with falsey values as long as it doesn't hurt.

File: PromisesObjC.podspec
Comment (ncooke3) [reply PromisesObjC.podspec:26]: Using `resource_bundle` is the correct path here. The issue with `resource[s]` surfaces when clients build the pod with _static linking_ enabled (`use_frameworks! :linkage => :static` in their Podfile). The pod's `PrivacyInfo.xcprivacy` gets thrown into the root of the app bundle, where it may collide with other manifests from the app or the app's other direct/transitive dependencies. Secondly– and I'm unsure if/how this will be enforced– the manifest won't be name spaced to the pod that owns it, so the app's resulting Privacy Report won't say _which_  SDK the information maps to. Examples below...

### App bundle when using `resource[s]`
```
PrivacyAppCocoaPods.app
├── Info.plist
├── PkgInfo
├── PrivacyAppCocoaPods
├── PrivacyInfo.xcprivacy   ← 💥 Will collide with other manifests. Not name spaced to pod.
└── _CodeSignature
    └── CodeResources
```

### App bundle when using `resource_bundle`
```
PrivacyAppCocoaPods.app
├── PromisesObjC_Privacy.bundle
│   ├── Info.plist
│   └── PrivacyInfo.xcprivacy   ← ✅ Not in the root of the app bundle. Name spaced to pod.
├── Info.plist
├── PkgInfo
├── PrivacyAppCocoaPods
└── _CodeSignature
    └── CodeResources
```

I suggest doing something like this:

```suggestion
  s.resource_bundle = {
    "#{s.name}_Privacy" => 'Sources/Resources/PrivacyInfo.xcprivacy'
  }
```

The above suggestion will support both the dynamic linking case and the static linking case.

File: Package.swift
Comment (ncooke3) [inline Package.swift:56]:
```suggestion
        .process("Sources/Resources/PrivacyInfo.xcprivacy")
```

nit: use `.process` instead of `.copy`. While I verified and it doesn't make a difference when generating the Privacy Report, [Apple's docs](https://developer.apple.com/documentation/PackageDescription/Resource/process(_:localization:)#:~:text=If%20possible%2C%20use%20this%20rule%20instead%20of) recommend `.process` over `.copy`.

File: PromisesSwift.podspec
Comment (ncooke3) [inline PromisesSwift.podspec:23]:
```suggestion
  s.resource_bundle = {
    "#{s.module_name}_Privacy" => 'Sources/Resources/PrivacyInfo.xcprivacy'
  }
```

File: Package.swift
Comment (ncooke3) [reply Package.swift:56]: Could you also please do this for the `Promises` target on line 85?

File: Sources/Resources/PrivacyInfo.xcprivacy
Comment (nabs-m) [reply Sources/Resources/PrivacyInfo.xcprivacy:12]: I interpreted it as iOS, etc only have to provide `NSPrivacyAccessedAPITypes` but not the other 3 🤷‍♂️ But you're right, it can have two meanings, so it doesn't hurt to play it safe.

File: PromisesObjC.podspec
Comment (nabs-m) [reply PromisesObjC.podspec]: @ncooke3 Super useful to know, thanks for the thorough explanation.

File: Package.swift
Comment (ykjchen) [reply Package.swift]: I assumed that since the `Promises` target depends on the `FBLPromises` target, the resource only needs to be included in the `FBLPromises` target. Is that correct?

Comment (ncooke3) [reply Package.swift]: I agree, and think this is likely fine. However, out of an abundance of caution, I think it's safer to include the manifest again for the `Promises` target. Since Apple's list calls out both the `FBLPromises` and `Promises` modules, I think both should explicitly have one (even if the contents are the same/empty). That way, if there is tooling to enforce any module on the list has a direct manifest (as opposed to a transitive one), all cases are covered.

Comment (ykjchen) [reply Package.swift]: Sounds good, added.

**Review by nabs-m — APPROVED** [review]

File: PromisesObjC.podspec
Comment (ncooke3) [inline PromisesObjC.podspec:27]: nit: Sorry, let's actually do module_name here to align with the other podspec.
```suggestion
    "#{s.module_name}_Privacy" => 'Sources/Resources/PrivacyInfo.xcprivacy'
```

**Comment by ykjchen** [comment]
Just waiting on internal approvals before merging.

**Comment by ncooke3** [comment]
Hi @ykjchen, there's one more fix to be merged into this branch: https://github.com/ykjchen/promises/pull/2

**Comment by nabs-m** [comment]
Any blockers left for this one? Trying to gauge when we can start consuming this in our app.

**Comment by ykjchen** [comment]
Internal processes still in progress, I'll bump the owners to see what the status is.

**Review by sksper1-stack — COMMENTED** [review]
[free-customized-LUTs.zip](https://github.com/user-attachments/files/22976080/free-customized-LUTs.zip)

**Pull request merged**
